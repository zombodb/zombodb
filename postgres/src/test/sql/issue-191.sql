CREATE TABLE "Issue191" (
  "ID"               SERIAL8 NOT NULL PRIMARY KEY,
  "BobIsYourUncle" TEXT
);

CREATE VIEW "Issue191_View" AS SELECT *, zdb('"Issue191"', ctid) FROM "Issue191";

CREATE INDEX idx_issue191
  ON "Issue191" USING zombodb (zdb('"Issue191"', ctid), zdb("Issue191")) WITH (url='http://localhost:9200/');

INSERT INTO "Issue191" ("BobIsYourUncle") VALUES ('abc');
INSERT INTO "Issue191" ("BobIsYourUncle") VALUES ('def');

--
-- one field from each of table and view
--
SELECT * FROM zdb_multi_search(ARRAY ['public."Issue191"'], NULL, ARRAY [ARRAY['BobIsYourUncle']], '');
SELECT * FROM zdb_multi_search(ARRAY ['public."Issue191_View"'], NULL, ARRAY [ARRAY['BobIsYourUncle']], '');

--
-- two fields from each of table and view
--
SELECT * FROM zdb_multi_search(ARRAY ['public."Issue191"'], NULL, ARRAY [ARRAY['ID', 'BobIsYourUncle']], '');
SELECT * FROM zdb_multi_search(ARRAY ['public."Issue191_View"'], NULL, ARRAY [ARRAY['ID', 'BobIsYourUncle']], '');

--
-- all fields from each of table and view
--
SELECT * FROM zdb_multi_search(ARRAY ['public."Issue191"'], NULL, NULL, '');
SELECT * FROM zdb_multi_search(ARRAY ['public."Issue191_View"'], NULL, NULL, '');

DROP TABLE "Issue191" CASCADE;