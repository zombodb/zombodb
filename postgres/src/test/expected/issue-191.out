CREATE TABLE "Issue191" (
  "ID"               SERIAL8 NOT NULL PRIMARY KEY,
  "BobIsYourUncle" TEXT
);
CREATE VIEW "Issue191_View" AS SELECT *, zdb('"Issue191"', ctid) FROM "Issue191";
CREATE INDEX idx_issue191
  ON "Issue191" USING zombodb (zdb('"Issue191"', ctid), zdb("Issue191")) WITH (url='http://localhost:9200/');
INSERT INTO "Issue191" ("BobIsYourUncle") VALUES ('abc');
INSERT INTO "Issue191" ("BobIsYourUncle") VALUES ('def');
--
-- one field from each of table and view
--
SELECT * FROM zdb_multi_search(ARRAY ['public."Issue191"'], NULL, ARRAY [ARRAY['BobIsYourUncle']], '');
 table_name | user_identifier | query | total | score |                                      row_data                                      
------------+-----------------+-------+-------+-------+------------------------------------------------------------------------------------
 "Issue191" |                 |       |     2 | {1,1} | [{"BobIsYourUncle":"abc","ctid":"(0,1)"}, {"BobIsYourUncle":"def","ctid":"(0,2)"}]
(1 row)

SELECT * FROM zdb_multi_search(ARRAY ['public."Issue191_View"'], NULL, ARRAY [ARRAY['BobIsYourUncle']], '');
   table_name    | user_identifier | query | total | score |                                     row_data                                     
-----------------+-----------------+-------+-------+-------+----------------------------------------------------------------------------------
 "Issue191_View" |                 |       |     2 | {1,1} | [{"_zdb_pkey":1,"BobIsYourUncle":"abc"}, {"_zdb_pkey":2,"BobIsYourUncle":"def"}]
(1 row)

--
-- two fields from each of table and view
--
SELECT * FROM zdb_multi_search(ARRAY ['public."Issue191"'], NULL, ARRAY [ARRAY['ID', 'BobIsYourUncle']], '');
 table_name | user_identifier | query | total | score |                                             row_data                                             
------------+-----------------+-------+-------+-------+--------------------------------------------------------------------------------------------------
 "Issue191" |                 |       |     2 | {1,1} | [{"ID":1,"BobIsYourUncle":"abc","ctid":"(0,1)"}, {"ID":2,"BobIsYourUncle":"def","ctid":"(0,2)"}]
(1 row)

SELECT * FROM zdb_multi_search(ARRAY ['public."Issue191_View"'], NULL, ARRAY [ARRAY['ID', 'BobIsYourUncle']], '');
   table_name    | user_identifier | query | total | score |                                            row_data                                            
-----------------+-----------------+-------+-------+-------+------------------------------------------------------------------------------------------------
 "Issue191_View" |                 |       |     2 | {1,1} | [{"_zdb_pkey":1,"ID":1,"BobIsYourUncle":"abc"}, {"_zdb_pkey":2,"ID":2,"BobIsYourUncle":"def"}]
(1 row)

--
-- all fields from each of table and view
--
SELECT * FROM zdb_multi_search(ARRAY ['public."Issue191"'], NULL, NULL, '');
 table_name | user_identifier | query | total | score |                                             row_data                                             
------------+-----------------+-------+-------+-------+--------------------------------------------------------------------------------------------------
 "Issue191" |                 |       |     2 | {1,1} | [{"ID":1,"BobIsYourUncle":"abc","ctid":"(0,1)"}, {"ID":2,"BobIsYourUncle":"def","ctid":"(0,2)"}]
(1 row)

SELECT * FROM zdb_multi_search(ARRAY ['public."Issue191_View"'], NULL, NULL, '');
   table_name    | user_identifier | query | total | score |                                                          row_data                                                          
-----------------+-----------------+-------+-------+-------+----------------------------------------------------------------------------------------------------------------------------
 "Issue191_View" |                 |       |     2 | {1,1} | [{"_zdb_pkey":1,"ID":1,"BobIsYourUncle":"abc","zdb":"(0,1)"}, {"_zdb_pkey":2,"ID":2,"BobIsYourUncle":"def","zdb":"(0,2)"}]
(1 row)

DROP TABLE "Issue191" CASCADE;
NOTICE:  drop cascades to view "Issue191_View"
